### PR TITLE
Improve example in Navigation/NavigationView

### PR DESCRIPTION
**Pull request by Tsuzat**
In a previous [update](https://github.com/bdlukaa/fluent_ui/pull/752), One can add/remove item(s) in NavigationPane but there was no implementation of this feature in the example application.
So, the following changes are made
- Add the PaneAction named *Add New Item* which adds a new item in NavigationPane.
- Refactor the code to achieve this, i.e. Moved `items` outside the `Widget Build` function and added `PaneAction` in footerItems.
- Refactor the code snippet according to implementation.

Before
Screenshot 2023-03-31 111442

After
ezgif-5-8749d1d32b

## Pre-launch Checklist

- [x] I have updated `CHANGELOG.md` with my changes  
- [x] I have run "dart format ." on the project  
- [x] I have added/updated relevant documentation

### CODE REVIEW

File: example/lib/screens/navigation/navigation_view.dart [hunk 1]
```diff
@@ -134,71 +194,31 @@ class _NavigationViewPageState extends State<NavigationViewPage>
...
+              items: items,
               footerItems: [
                 PaneItem(
                   icon: const Icon(FluentIcons.settings),
                   title: const Text('Settings'),
                   body: const _NavigationBodyItem(),
                 ),
+                PaneItemAction(
+                  icon: const Icon(FluentIcons.add),
+                  title: const Text("Add New Item"),
```
Comment (bdlukaa) [inline example/lib/screens/navigation/navigation_view.dart:206]:
```suggestion
                  title: const Text('Add New Item'),
```

File: example/lib/screens/navigation/navigation_view.dart [hunk 2]
```diff
@@ -134,71 +194,31 @@ class _NavigationViewPageState extends State<NavigationViewPage>
...
+                  onTap: () {
+                    items.add(
+                      PaneItem(
+                        icon: const Icon(FluentIcons.new_folder),
+                        title: const Text("New Item"),
+                        body: const Center(
+                          child: Text(
+                            "This is a newly added Item",
```
Comment (bdlukaa) [inline example/lib/screens/navigation/navigation_view.dart:214]:
```suggestion
                        title: const Text('New Item'),
                        body: const Center(
                          child: Text(
                            'This is a newly added Item',
```

File: example/lib/screens/navigation/navigation_view.dart [hunk 3]
```diff
@@ -228,42 +248,105 @@ class _NavigationViewPageState extends State<NavigationViewPage>
...
+        title: const Text("Add New Item"),
+        onTap: () {
+          // Your Logic to Add New `NavigationPaneItem`
+          items.add(
+            PaneItem(
+              icon: const Icon(FluentIcons.new_folder),
+              title: const Text("New Item"),
+              body: const Center(
+                child: Text(
+                  "This is a newly added Item",
```
Comment (bdlukaa) [inline example/lib/screens/navigation/navigation_view.dart:343]:
```suggestion
              title: const Text('New Item'),
              body: const Center(
                child: Text(
                  'This is a newly added Item',
```

**Review by bdlukaa — CHANGES_REQUESTED** [review]

**Comment by Tsuzat** [comment]
Hey, I have made changes and amended the commit. Also, double quotes have been changes to single quotes in Code Snipppet too.

**Review by bdlukaa — APPROVED** [review]

**Pull request merged**